app/grandchallenge/archives/migrations/0017_archive_item_delete_permission_granting.py

```diff
@@ -0,0 +1,45 @@
+from django.db import migrations
+
+
+def add_archive_item_delete_permissions(apps, schema_editor):
+    Archive = apps.get_model("archives", "Archive")  # noqa: N806
+    ArchiveItem = apps.get_model("archives", "ArchiveItem")  # noqa: N806
+    ArchiveItemGroupObjectPermission = apps.get_model(  # noqa: N806
+        "archives", "ArchiveItemGroupObjectPermission"
+    )
+    Permission = apps.get_model("auth", "Permission")  # noqa: N806
+
+    queryset = Archive.objects.order_by("-created").prefetch_related(
+        "editors_group", "items"
+    )
+    if queryset:

```

Comment:
Doing `if queryset` will execute the queryset and fetch all the archives and archive items, will be very slow as it would fetch a tonne of data from the db at once. Using `exists()` will be much faster and keeps the advantage of doing the iterator later:

```suggestion
    if queryset.exists():

```